quickwit-backward-compat/tests/split_metadata_non_regression.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+
+use anyhow::Context;
+use once_cell::sync::OnceCell;
+use quickwit_metastore::{SplitMetadata, SplitMetadataAndFooterOffsets, SplitState};
+
+fn check_version(
+    version: usize,
+    split_metadata: &SplitMetadataAndFooterOffsets,
+) -> anyhow::Result<()> {
+    // Feel free to enrich this function if we start introducing default values that are not
+    // deterministic. (e.g. depends on time.)
+    static EXPECTED_DESERIALIZED_SPLIT_METADATA: OnceCell<
+        HashMap<usize, SplitMetadataAndFooterOffsets>,
+    > = OnceCell::new();
+    let expected_split_metadata = EXPECTED_DESERIALIZED_SPLIT_METADATA
```

Comment:
I would remove the once cell and pass it as parameter from `test_backward_compatibility`

Reply:
change the logic to check against a expected.json file.